src/OnboardingSPA/components/Sidebar/components/Customize/DesignFontsPanel/index.js

```diff
@@ -337,6 +344,53 @@ const DesignFontsPanel = forwardRef(
 			...new Set( [ ...fontsHeading, ...fontsContent ] ),
 		];
 
+		const fontOptionsNames = [
+			...new Set( [ ...fontsHeadingNames, ...fontsContentNames ] ),
+		];
+
+		useEffect( () => {
+			// Create a div element for the invisible text
+			const invisibleTextContainer = document.createElement( 'div' );
+			invisibleTextContainer.style.position = 'absolute';
+			invisibleTextContainer.style.top = '-9999px';
+			invisibleTextContainer.style.left = '-9999px';
+			invisibleTextContainer.style.visibility = 'hidden';
+
+			// Append the invisible text container to the body
+			document.body.appendChild( invisibleTextContainer );
+
+			// Apply all fonts to the invisible text
+			fontOptionsNames.forEach( ( font ) => {
+				const invisibleText = document.createElement( 'span' );
+				invisibleText.textContent =
+					'Hidden text to check font availability';
+				invisibleText.style.font = `1em ${ font }`;
+				invisibleTextContainer.appendChild( invisibleText );
+			} );
+
+			const checkFontsLoaded = setInterval( () => {
+				if ( areFontsLoaded() ) {
+					clearInterval( checkFontsLoaded );
+					setFontsLoaded( true );
+				}
+			}, 100 );
+
+			return () => {
+				clearInterval( checkFontsLoaded );
+			};
+			// eslint-disable-next-line react-hooks/exhaustive-deps
+		}, [ fontOptions ] );
+
```

Comment:
@mr-vara we should avoid rendering hidden data unless absolutely necessary. The correct approach would be to wait for the fonts API to complete and then removing the skeleton. If the fonts are not loaded this seems a bigger issue with the Wonder Theme assets itself. Can we verify if this is necessary even after the expedite of the Wonder Theme installation and activation?